alacritty_terminal/src/renderer/mod.rs

```diff
@@ -719,6 +719,7 @@ impl QuadRenderer {
         visual_bell_color: Rgb,
         visual_bell_intensity: f64,
         cell_line_rects: Vec<RenderRect>,
+        message_bar_rect: Option<RenderRect>,
```

Comment:
Currently this is just a seemingly generic function that takes a bunch of very specific arguments. Like passing `visual bell` to it doesn't really seem super related to `draw_rects` in the more generic sense.

Do you think it would make sense to have the `draw_rects` be purely about drawing `Vec<RenderRect>`?

We could then just call it multiple times in the desired order at callsite.

A separate `draw_visual_bell` method could take the visual bell parameters and convert it to a `RenderRect`.

I think generally this sounds like a good idea to me, though one concern might be that we're switching program/blend funcs/data buffers a lot then. But that would likely not make any difference in practice? Another thing might be that taking a `Vec` is suboptimal, though taking just `RenderRect` might complicate the situation at the callsite.

Alternatively we could keep the function pretty much identical, but having a strict rendering order with our rects. We can convert the visual bell separately and then push it onto our `rect` buffer in the correct order. Then this function would also take just a `Vec<RenderRect>` and do all its name says it does. I've kinda come up with this idea while writing this review, but it sounds like the better solution to me. What do you think?

Reply:
Oh, let me try something